crates/matching-engine/src/cfmm/uniswap/mod.rs

```diff
@@ -1,8 +1,32 @@
+use alloy_primitives::{aliases::I24, I256};
+
 pub mod pool;
+pub mod pool_data_loader;
 pub mod pool_manager;
 pub mod pool_providers;
 pub mod tob;
 
+fn i32_to_i24(val: i32) -> I24 {
+    let mut bytes = [0u8; 3];
+    let value_bytes = val.to_be_bytes();
+    bytes[..].copy_from_slice(&value_bytes[1..]);
+    I24::from_be_bytes(bytes)
+}
```

Comment:
so i32 is stored in twos compliment right so i don't think we can just cut off  a byte like this as it will most likely be using all bytes even tho it can fit into a i24

Reply:
also we probably want to add a panic here incase the number is to big